gnd/src/main/java/com/google/android/gnd/repository/FeatureRepository.java

```diff
@@ -50,29 +51,24 @@ public FeatureRepository(
       ProjectRepository projectRepository,
       DataSyncWorkManager dataSyncWorkManager) {
     this.localDataStore = localDataStore;
+    this.remoteDataStore = remoteDataStore;
     this.projectRepository = projectRepository;
     this.dataSyncWorkManager = dataSyncWorkManager;
-
-    streamFeaturesToLocalDb(remoteDataStore);
   }
 
   /**
-   * Mirrors features in the current project from the remote db into the local db when the network
+   * Mirrors features in the specified project from the remote db into the local db when the network
    * is available. When invoked, will first attempt to resync all features from the remote db,
    * subsequently syncing only remote changes.
    */
-  private void streamFeaturesToLocalDb(RemoteDataStore remoteDataStore) {
-    // TODO: Move to Application or background service.
-    // TODO: Is this even working? If the returned Disposable is garbage collected this will be
-    // interrupted.
-    projectRepository
-        .getActiveProjectOnceAndStream()
-        .compose(Loadable::values)
-        .switchMap(remoteDataStore::loadFeaturesOnceAndStreamChanges)
+  public Completable syncFeatures(Project project) {
+    return remoteDataStore
+        .loadFeaturesOnceAndStreamChanges(project)
         .switchMap(event -> updateLocalFeature(event).toFlowable())
-        .subscribe();
+        .ignoreElements();
```

Comment:
Neat! Didn't know about this one. It's a nice signal that we only care about side effects here...but do we actually need it? Converting the completable to the flowable I *think* does effectively the same thing? http://reactivex.io/RxJava/javadoc/io/reactivex/Completable.html#toFlowable-- -- since the completable already only emits terminal events.

Reply:
Trying to simplify this... PTAL!